driver-mapping/src/main/java/com/datastax/driver/mapping/Mapper.java

```diff
@@ -128,6 +128,9 @@ public void onSuccess(PreparedStatement stmt) {
                     @Override
                     public void onFailure(Throwable t) {
                         future.setException(t);
+                        // do not keep a failed future in the query cache
+                        preparedQueries.remove(pqk, future);
+                        logger.error("Query preparation failed: " + queryString, t);
```

Comment:
minor: couldn't we log before setting the exception on the future or does it not matter?